Feat: Add HasLegendSupport helper method (#66)
camptocamp/geospatial-sdk: packages/legend/README.md

});
```

### Checking legend support

> Use `hasLegendSupport` to check whether a layer can have a legend before attempting to render one. It is a type guard: narrowing the layer to `MapContextLayerWms | MapContextLayerWmts`.

Reviewer: ```suggestion
Use `hasLegendSupport` to check whether a layer can have a legend before attempting to render one.
```

too specific

```typescript
import {
  hasLegendSupport,
  createLegendFromLayer,
} from "@geospatial-sdk/legend";

if (hasLegendSupport(layer)) {
  // layer is narrowed to MapContextLayerWms | MapContextLayerWmts
  createLegendFromLayer(layer).then((legendDiv) => {
    if (legendDiv) document.body.appendChild(legendDiv);
  });
}
```

> Note: `hasLegendSupport` is a type-level check only. A WMTS layer may still return `null` from `createLegendFromLayer` if no legend URL is declared in its capabilities.

Reviewer: ```suggestion
Note: `hasLegendSupport` is a type-level check only. Legend generation might still return `null` for other reasons.
```
